From ee709595b491447c9c92f5e4b096835b48ca5dff Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Wed, 26 Feb 2025 10:08:11 +0100
Subject: [PATCH] arm64: dts: st: add support of continuous display on
 stm32mp235 DK board

Set the backlight, the panel, ltdc & lvds on at probe.

Change-Id: I4683108c474039d0722720e146bfe14df23d2afe
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/440025
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp235f-dk.dts | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp235f-dk.dts b/arch/arm64/boot/dts/st/stm32mp235f-dk.dts
index 6bc923b7f28f..c07c12d8a876 100644
--- a/arch/arm64/boot/dts/st/stm32mp235f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp235f-dk.dts
@@ -35,6 +35,8 @@ chosen {
 
 		framebuffer {
 			compatible = "simple-framebuffer";
+			clocks = <&rcc CK_BUS_LTDC>, <&rcc CK_KER_LTDC>,
+				 <&rcc CK_BUS_LVDS>, <&rcc CK_KER_LVDSPHY>;
 			status = "disabled";
 		};
 	};
@@ -136,6 +138,7 @@ panel_lvds: panel-lvds {
 		compatible = "edt,etml0700z9ndha", "panel-lvds";
 		enable-gpios = <&gpioi 4 GPIO_ACTIVE_HIGH>;
 		backlight = <&panel_lvds_backlight>;
+		default-on;
 		status = "okay";
 
 		width-mm = <156>;
@@ -165,7 +168,7 @@ panel_lvds_backlight: panel-lvds-backlight {
 		compatible = "gpio-backlight";
 		gpios = <&gpioi 7 GPIO_ACTIVE_HIGH>;
 		default-on;
-		default-brightness-level = <0>;
+		default-brightness-level = <1>;
 		status = "okay";
 	};
 
@@ -485,6 +488,7 @@ timer {
 };
 
 &ltdc {
+	default-on;
 	status = "okay";
 
 	port {
@@ -504,9 +508,11 @@ ltdc_ep1_out: endpoint@1 {
 };
 
 &lvds {
-	status = "okay";
+	default-on;
 	vdd-supply =  <&scmi_vddcore>;
 	vdda18-supply = <&scmi_v1v8>;
+	status = "okay";
+
 	ports {
 		#address-cells = <1>;
 		#size-cells = <0>;
-- 
2.47.2

